src/vs/workbench/services/environment/browser/environmentService.ts

```diff
@@ -115,7 +115,7 @@ export class BrowserWorkbenchEnvironmentService implements IWorkbenchEnvironment
 	get snippetsHome(): URI { return joinPath(this.userRoamingDataHome, 'snippets'); }
 
 	@memoize
-	get userDataSyncHome(): URI { return joinPath(this.userRoamingDataHome, 'sync'); }
+	get userDataSyncHome(): URI { return joinPath(this.userRoamingDataHome, 'sync', this.options.workspaceId); }
```

Comment:
Keep in mind that `this.options.workspaceId` seems to be `empty-window` for windows without workspace, so you might end up with the same `userDataSyncHome` for multiple windows.

Reply:
Of course, you can end up with the same situation when a user opens the same tab multiple times.

Reply:
Yes, I am aware that empty windows will have same workspace id and as you said it is similar to opening same workspace in multiple tabs. But Sync is capable of handling such cases - Multiple windows are sharing same user data and extensions. This is to fix the other way around scenario - where multiple workspaces are not sharing extensions and sharing user data. In that case sync can cause randomness like - removes extensions that other sync added - MicrosoftDocs/vsonline#448

Ideally making sync state scoped per window is ideal but that does not reciprocate general workspaces in web and also adds an overhead of not having initial sync state when same workspace is opened in a different window which shows conflicts while syncing.

Hence took the middle ground of scoping sync state per workspace and making sure sync can handle the case same workspace is opened in multiple windows.